packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -101,10 +104,24 @@ class SnackBarAction extends StatefulWidget {
   /// hovered and others.
   final Color? textColor;
 
+  /// The button background fill color. If not provided, defaults to
+  /// [SnackBarThemeData.actionBackgroundColor].
+  ///
+  /// If [backgroundColor] is a [MaterialStateColor], then the text color will
```

Comment:
I'd consider adding an extra line of documentation explaining that if `backgroundColor` is a `MaterialStateColor`, then any value of `disabledBackgroundColor` won't be used.  We could possibly add a formal assertion that `disabledBackgroundColor` shouldn't be provided if `backgroundColor` is present and is a `MaterialStateColor`; this would have to be done here and in the `SnackBarTheme`.